tests/src/test/scala-3/fs2/dom/Assertions.scala

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2022 Arman Bilge
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fs2.dom
+
+import org.scalajs.dom
+import scala.scalajs.js
+import js.annotation.*
+import scala.quoted.*
+
+object Assertions {
+  private def assertGlobalImpl[A <: js.Any](
+      loc: Expr[munit.Location]
+  )(using Type[A], Quotes): Expr[Unit] = {
```

Comment:
Sorry to be stupid: what does this macro do and why does it need to be a macro?

At risk of being grumpy, I'd lean towards having no test at all rather than too much macros 😅

Reply:
Thanks for the detailed explanation!

> adding an implicit dependency on a specific browser driver didn't feel right

Sorry I missed this, why doesn't this feel right? We already have browser tests so nothing special/extra would be needed here.

Reply:
I think it's fine if it's for the more conservative features, but I was worried that if we're adding more recent API's, the browser version on the CI/distro could lag behind and I remember working with travis CI when their ubuntu distro was several LTS behind and it was really painful.

Maybe I'm misunderstanding how the tests work? I was under the impression that they're using the version of firefox and chrome on the path?

Reply:
I haven't experienced any issues like that. GitHub keeps the software on the runners up-to-date. For example in http4s-dom we are using the latest Fetch streaming request capabilities only added in recent versions of Chrome.

If that fails, we can run our CI with Nix like we do in http4s.

Reply:
In which case, I'll switch over to testing the globals exist then?

Reply:
Basically all I was imagining was a test that creates the element in the DOM and then does an `instanceOf` test to verify it got back what we expected.

Reply:
Don't even think we need to create it (I don't think we can for some elements, like the `AudioNode` stuff). I just need to get the global object and make sure it exists as a symbol?

Reply:
@armanbilge

I've switched to using a method instead.

Reply:
Should probably move this to `scala` rather than only `scala-3`.

Reply:
Great, thank you for changing that!!